coda/coda_mdstore/tests/test_utils.py

```diff
@@ -1,14 +1,573 @@
+
+        assert bag.bagit_version == ''
+        assert error is None
+
+    def test_bagitVersion_is_set(self, bag_xml):
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert bag.bagit_version == str(bag_xml.bagitVersion)
+        assert error is None
+
+    def test_baggingDate_not_set(self, bag_xml):
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+
+        assert isinstance(bag.bagging_date, datetime)
+        assert error is None
+
+    def test_has_bag_info_objects(self, bag_xml):
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+        assert len(bag_infos) == 2
+        # Verify that all of the bag_infos are instances of models.Bag_Info
+        assert all([isinstance(m, models.Bag_Info) for m in bag_infos])
+
+    def test_has_no_bag_info_objects(self, bag_xml):
+        del bag_xml.bagInfo.item
+        del bag_xml.bagInfo.item
+        bag, bag_infos, error = presentation.xmlToBagObject(bag_xml)
+        assert len(bag_infos) == 0
+
+
+@pytest.fixture
+def bag_xml():
+    xml = """<?xml version="1.0"?>
+        <entry xmlns="http://www.w3.org/2005/Atom">
+        <title>ark:/67531/coda2</title>
+        <id>ark:/67531/coda2</id>
+        <updated>2013-06-05T17:05:33Z</updated>
+        <author>
+            <name>server</name>
+        </author>
+        <content type="application/xml">
+            <bag:codaXML xmlns:bag="http://digital2.library.unt.edu/coda/bagxml/">
+            <bag:name>ark:/67531/coda2</bag:name>
+            <bag:fileCount>43</bag:fileCount>
+            <bag:payloadSize>46259062</bag:payloadSize>
+            <bag:bagitVersion>0.96</bag:bagitVersion>
+            <bag:lastVerified/>
+            <bag:bagInfo>
+                <bag:item>
+                    <bag:name>Payload-Oxum</bag:name>
+                    <bag:body>46259062.43</bag:body>
+                </bag:item>
+                <bag:item>
+                    <bag:name>Tag-File-Character-Encoding</bag:name>
+                    <bag:body>UTF-8</bag:body>
+                </bag:item>
+            </bag:bagInfo>
+            </bag:codaXML>
+        </content>
+        </entry>
+    """
+    return objectify.fromstring(xml)
+
+
+@pytest.mark.django_db
+@pytest.mark.usefixture('bag_xml')
+class TestCreateBag:
+    """
+    Tests for coda_mdstore.presentation.createBag.
+    """
+
+    @pytest.mark.xfail(reason='Exception in function will never be raised.')
+    def test_raises_exception_when_xml_cannot_be_parsed(self):
+        with pytest.raises(Exception) as e:
+            presentation.createBag('')
+        assert str(e) == 'Unable to parse uploaded XML'
+
+    @pytest.mark.xfail(reason='Exception in function will never be raised.')
+    def test_raises_exception_when_content_element_not_present(self):
+        with pytest.raises(Exception) as e:
+            presentation.createBag('<root/>')
+        assert str(e) == 'No content to parse uploaded XML'
```

Comment:
It is a test for this exception

[Code](https://github.com/unt-libraries/coda/blob/master/coda/coda_mdstore/presentation.py#L172)